5 Arrays and Cycles.js

```diff
@@ -1,90 +1,91 @@
 
-let arr = ["Olga", "Roman", "Yana", "Oleg"];
-let result = arr.map(function (item, index, array) {
+let names = ["Olga", "Roman", "Yana", "Oleg"];
+let result = names.map(function (item, index, array) {
   return "hello " + item;
 });
 console.log(result);
 
 /*Дан массив состоящий из названий фильмов, выполните перебор массива 
 с выводом в консоль названия каждого фильма*/
 
-let arr = ["matrix", "die hard", "snatch", "shrek 3"];
-for (let i = 0; i < arr.length; i++) {
-  console.log(arr[i]);
+let movies = ["matrix", "die hard", "snatch", "shrek 3"];
+for (let i = 0; i < movies.length; i++) {
+  console.log(movies[i]);
 }
 
 //Дан массив производителей автомобилей, преобразовать массив в строку и обратно в массив
 
-let arr = ["audi", "bmw", "fiat", "lada"];
-let str = arr.join(",");
+let manufactures = ["audi", "bmw", "fiat", "lada"];
+let str = manufactures.join(",");
 console.log(str);
-arr = str.split(",");
-console.log(arr);
+manufactures = str.split(",");
+console.log(manufactures);
 
 // Преобразовать числовой массив в Boolean
-let arr = [1, 0, 7, 8, 0, 0, 5, 6];
-let result = arr.map(function (item, index, array) {
+let numbersToBoolean = [1, 0, 7, 8, 0, 0, 5, 6];
+let boolArray = numbersToBoolean.map(function (item, index, array) {
   return Boolean(item);
 });
-console.log(result);
+console.log(boolArray);
 
 //Отфильтровать массив [1,6,7,8,3,4,5,6] по значению >3
 
-let arr = [1, 6, 7, 8, 3, 4, 5, 6];
-let result = [];
-for (let i = 0; i < arr.length; i++) {
-  if (arr[i] > 3) {
-    result.push(arr[i]);
-  }
-}
-console.log(result);
+let numbersToFilter = [1, 6, 7, 8, 3, 4, 5, 6];
+let filtered = numbersToFilter.filter((n) => n > 3);
+console.log(filtered);
 
 //Отсортировать массив [1,6,7,8,3,4,5,6] по убыванию
-let arr = [1, 6, 7, 8, 3, 4, 5, 6];
-arr.sort(function (a, b) {
+let decreaseNumbers = [1, 6, 7, 8, 3, 4, 5, 6];
+decreaseNumbers.sort(function (a, b) {
   return b - a;
 });
-console.log(arr);
+console.log(decreaseNumbers);
 
 //Написать функцию, которая принимает два параметра - массив и число и выводит индекс элемента массива равный числу
 
-function callArray(arr, n) {
-  for (let i = 0; i < arr.length; i++) {
-    if (arr[i] == n) {
+function callArray(indexArray, n) {
+  for (let i = 0; i < indexArray.length; i++) {
+    if (indexArray[i] === n) {
       return i;
-      break;
     }
   }
 }
 console.log(callArray([8, 7, 6, 6, 5, 4, 3, 1], 6));
 
 //Реализовать цикл, который будет выводить число а, пока оно не станет меньше 10
 
-let a = prompt("Enter a");
-while (a < 10) {
-  console.log(a);
-  a = prompt("Enter next value");
+function lessThan10(a) {
+  while (a >= 10) {
+    console.log(a);
+    a--;
+  }
 }
 
 //Реализовать цикл, который выводит в консоль простые числа
 
-console.log(1);
-let n = 3;
-nextPrime: for (let i = 2; i <= n; i++) {
-  for (let j = 2; j < i; j++) {
-    if (i % j == 0) continue nextPrime;
+function isPrimeNumber(x) {
+  if (x === 2) return x;
+  for (n = 2; n < x; n++) {
```

Comment:
variable `n` should be declared in loop conditions.

Reply:
Just declared n and run this function. Either declaring or not, function works identically. Is it expected behavior?

Reply:
Expected, but we always should to declare variables.